client/state/timezones/test/reducer.js

```diff
@@ -0,0 +1,335 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import { useSandbox } from 'test/helpers/use-sinon';
+
+import timezonesReducer, {
+	byContinentsReducer,
+	labelsReducer,
+	rawOffsetReducer,
+} from '../reducer';
+
+import { timezonesReceive } from '../actions';
+import { fromApi } from 'state/data-layer/wpcom/timezones';
+
+/**
+ * Fixture data
+ */
+import {
+	RAW_OFFSETS,
+	TIMEZONES,
+	TIMEZONES_BY_CONTINENT,
+} from './fixture';
+
+describe( 'reducer', () => {
+	let sandbox;
+
+	useSandbox( newSandbox => {
+		sandbox = newSandbox;
+		sandbox.stub( console, 'warn' );
+	} );
+
+	it( 'should export expected reducer keys', () => {
+		expect( timezonesReducer( undefined, {} ) ).to.have.keys( [
+			'byContinents',
+			'labels',
+			'rawOffsets',
+		] );
+	} );
+
+	describe( '#rawOffsetReducer()', () => {
+		it( 'should default to an empty Object', () => {
+			expect( rawOffsetReducer( undefined, {} ) ).to.eql( {} );
+		} );
+
+		it( 'should index timezones.rawOffsets state', () => {
+			const initialState = undefined;
+			const restAPIResponse = {
+				manual_utc_offsets: RAW_OFFSETS,
+				timezones: TIMEZONES,
+				timezones_by_continent: TIMEZONES_BY_CONTINENT,
+			};
+
+			const apiResponse = fromApi( restAPIResponse );
+			const action = timezonesReceive( apiResponse );
+
+			const expectedState = apiResponse.rawOffsets;
+			const newState = rawOffsetReducer( initialState, action );
+			expect( newState ).to.eql( expectedState );
+		} );
+
+		it( 'should override timezones.rawOffsets state', () => {
+			const initialState = fromApi( {
+				manual_utc_offsets: [
+					{ value: 'foo', label: 'bar' }
+				],
+				timezones: TIMEZONES,
+				timezones_by_continent: TIMEZONES_BY_CONTINENT,
+			} ).rawOffsets;
+
+			deepFreeze( initialState );
+
+			const restAPIResponse = {
+				manual_utc_offsets: RAW_OFFSETS,
+				timezones_by_continent: TIMEZONES_BY_CONTINENT,
+			};
+
+			const apiResponse = fromApi( restAPIResponse );  // data-layer processing
+			const action = timezonesReceive( apiResponse );
+
+			const expectedState = apiResponse.rawOffsets;
+			const newState = rawOffsetReducer( initialState, action );
+			expect( newState ).to.eql( expectedState );
+		} );
+
+		it( 'should persist state', () => {
+			const initialState = fromApi( {
+				manual_utc_offsets: RAW_OFFSETS,
+				timezones: TIMEZONES,
+				timezones_by_continent: TIMEZONES_BY_CONTINENT,
+			} );
+			deepFreeze( initialState );
+
+			const action = { type: 'SERIALIZE' };
+
+			const expectedState = initialState;
+			const newState = rawOffsetReducer( initialState, action );
+			expect( newState ).to.eql( expectedState );
+		} );
```

Comment:
the integration part is that we have made a promise in the persistence layer: "if you provide a schema and use something like `createReducer()` or `withSchema()` then we will handle the persistence"

if we're going to provide duplicate tests that that layer does its job everywhere we rely on it then we'll be adding tons and tons of redundant code to maintain as well as spread out the focus from that layer. that is to say, if we kept these as unit tests and trusted each _unit_ to do its job, then we could focus our efforts on the persistence layer to find and fix bugs in one place instead of spread ourselves out on each different piece that depends on it.